test/boost/secondary_index_test.cc

```diff
@@ -1575,6 +1575,228 @@ SEASTAR_TEST_CASE(test_token_order) {
     });
 }
 
+SEASTAR_TEST_CASE(test_select_with_token_range_cases) {
+    return do_with_cql_env_thread([] (auto& e) {
+        cquery_nofail(e, "CREATE TABLE t (pk int, v int, PRIMARY KEY(pk))");
+
+        for (int i = 0; i < 7; i++) {
+            // v=1 in each row, so WHERE v = 1 will select them all
+            cquery_nofail(e, format("INSERT INTO t (pk, v) VALUES ({}, 1)", i).c_str());
+        }
+
+        auto get_result_rows = [&](int start, int end) {
+            std::vector<std::vector<bytes_opt>> result;
+            for (int i = start; i <= end; i++) {
+                result.push_back({ testset_pks[i][0], int32_type->decompose(1) });
+            }
+            return result;
+        };
+
+        auto get_query = [&](sstring token_restriction) {
+            return format("SELECT * FROM t WHERE v = 1 AND {} ALLOW FILTERING", token_restriction);
+        };
+
+        auto q = [&](sstring token_restriction) {
+            return cquery_nofail(e, get_query(token_restriction).c_str());
+        };
+
+        auto inclusive_inclusive_range = [](int64_t start, int64_t end) { return format("token(pk) >= {} AND token(pk) <= {}", start, end); };
```

Comment:
I have here the same question I had earlier. All these test cases have token(pk) on the **left hand side** of the inequality. Does the CQL language forbid having it in the right hand side, e.g., "0 < token(pk)"? Does Cassandra also forbid it (here is where cql-pytests are better than those C++ tests...)?
If it's forbidden, we need a test demonstrating it's forbidden (and if you write a C++ test, please at least test manually on Cassandra). But if it's not forbidden, we need tests where token(pk) is on the right hand side (and they will probably fail because your implementation gave the left hand side special treatment).